tests/test_auth.py

```diff
@@ -152,14 +153,42 @@ def test_get_username_runtime_error_suppressed(
     entered_username, keyring_no_backends_get_credential, caplog, config
 ):
     assert auth.Resolver(config, auth.CredentialInput()).username == "entered user"
-    assert caplog.messages == ["fail!"]
+    assert re.search(
+        r"Error from keyring.+Traceback.+RuntimeError: fail!", caplog.text, re.DOTALL
+    )
 
 
 def test_get_password_runtime_error_suppressed(
     entered_password, keyring_no_backends, caplog, config
 ):
     assert auth.Resolver(config, auth.CredentialInput("user")).password == "entered pw"
-    assert caplog.messages == ["fail!"]
+    assert re.search(
+        r"Error from keyring.+Traceback.+RuntimeError: fail!", caplog.text, re.DOTALL
+    )
+
+
+@pytest.mark.parametrize(
+    "get_credential",
+    [
+        auth.Resolver.get_username_from_keyring,
+        auth.Resolver.get_password_from_keyring,
+    ],
```

Comment:
After poking around a bit, I think these should be treated as private methods, and we should test `auth.Resolver.username` and `.password`, as done in the previous two tests.